### PR TITLE
cython3 compatibility

### LINKED ISSUE #4019: Check cython compilation warnings
There's some `performance hint` that needs to be addressed, e.g. :

```
performance hint: src/silx/math/_colormap.pyx:348:46: Exception check will always require the GIL to be acquired. Declare the function as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.
```

### PR DESCRIPTION
**Pull request by kif**
Remove compilation warnings
close #4019

### CODE REVIEW

File: src/silx/image/marchingsquares/_mergeimpl.pyx [hunk 1]
```diff
@@ -1,5 +1,11 @@
+#cython: embedsignature=True, language_level=3
+## This is for optimisation
+#cython: boundscheck=False, wraparound=False, cdivision=True, initializedcheck=False,
```
Comment (t20100) [inline src/silx/image/marchingsquares/_mergeimpl.pyx:3]: Why is this is needed globally?

File: src/silx/math/_colormap.pyx [hunk 2]
```diff
@@ -1,3 +1,8 @@
+#cython: embedsignature=True, language_level=3
+## This is for optimisation
+##cython: boundscheck=False, wraparound=False, cdivision=True, initializedcheck=False,
```
Comment (t20100) [inline src/silx/math/_colormap.pyx:3]: That's disabled, right?

File: src/silx/math/_colormap.pyx [hunk 3]
```diff
@@ -101,7 +106,7 @@ ctypedef fused image_types:
 
 # Normalization
 
-ctypedef double (*NormalizationFunction)(double) nogil
+# ctypedef double (*NormalizationFunction)(double) nogil
```
Comment (t20100) [inline src/silx/math/_colormap.pyx:109]: This is no longer a valid syntax?

**Review by t20100 — APPROVED** [review]
LGTM, thanks!

File: src/silx/math/_colormap.pyx (hunk 3 above)
Comment (kif) [reply src/silx/math/_colormap.pyx:109]: It is still valid but it is unused ... so I commented it to ensure nothing breaks.
To be removed once we are sure it makes no regression anywhere.

**Pull request merged**